src/data/finset/basic.lean

```diff
@@ -251,6 +251,14 @@ by rw [coe_singleton, set.singleton_subset_iff]
   {a} ⊆ s ↔ a ∈ s :=
 singleton_subset_set_iff
 
+/-! ### cons -/
+
+def cons {α} (a : α) (s : finset α) (h : a ∉ s) : finset α :=
```

Comment:
What's the relation between this and `insert`? I guess you want some feature of the underlying representation here that doesn't hold for `insert`? Could you clarify this in the doc string, and for `mem_cons`/`mem_insert` below?

Reply:
This is now covered in the doc string. The short answer is it doesn't require `decidable_eq`, and it also has better defeqs in case you want to force the list component of it.